### DISCUSSION TITLE
`empty` deprecated in `Record`, but not in `ReadonlyRecord`

### DISCUSSION THREAD

**Discussion by fmpanelli** [gcanti/fp-ts#1578] [Q&A] [answered]
`Record` and `ReadonlyRecord` are of course very similar.

I noticed that `empty` is deprecated in `Record`:

```ts
/**
 * Use a new `{}` instead.
 *
 * @since 2.0.0
 * @deprecated
 */
export const empty: Record<string, never> = {}
```

But it is not deprecated in `ReadonlyRecord`:

```ts
/**
 * @since 2.5.0
 */
export const empty: ReadonlyRecord<string, never> = {}
```

I am not sure if this is on purpose or I should raise an issue and propose a PR.

**Comment by gcanti** [comment, marked as answer]
@fmpanelli this is on purpose, `empty` in `Record.ts` is **mutable**, it shouldn't be used (hence the deprecation).

**Reply by DenisFrezzato** [reply]
Then I suggest to mention the reason why it's deprecated. What do you think?

**Reply by fmpanelli** [reply]
Giulio and Denis, I see your point and I would like to double-down on it.

And after experimenting a little bit I was able to write pretty indecent code also with `readonlyRecord.empty`.
Please, have a look at this crap of mine:
```ts
// Please! Don't do this at home!
function f(r: Record<string, unknown>) {
  r.a = 3;
}
const e = readonlyRecord.empty;
assert.deepStrictEqual(readonlyRecord.empty, {});
f(e);
assert.deepStrictEqual(readonlyRecord.empty, { a: 3 });
```

Maybe `readonlyRecord.empty` should also be _considered harmful_ and deprecated?
